### PR TITLE
feat(validation): add migration-chain validation policy

### PR DESCRIPTION
**Pull request by dimileeh**
Automatically opened by AWF workspace `ws_f1f1f57ee64e470a9ee44821` (agent: `codex`, model: `gpt-5.5`, effort: `xhigh`).

**External task ID**: pre-gke-p1-alembic-migration-chain-validation-20260430T0948Z

### Task
Implement a narrow P1 Python/Alembic workload slice: add migration-chain validation for Python/Alembic profiles. Use strict TDD: first add failing tests for detecting multiple heads, missing down_revision chains, duplicate revision ids, branch-label anomalies, and the clean single-head case. Then expose a profile validation command/policy helper that can be enabled by Python DB-backed profiles and emits structured reason codes/logs. Integrate with existing Alembic resolver/target-branch monitor only as a reusable validation policy, not a new auto-fix flow. Do not lower coverage or quality gates.

---
Validation: 8 profile command(s) passed inside the workspace container.

### CODE REVIEW

File: src/awf/runtime/validation.py [hunk 1]
```diff
@@ -409,6 +448,79 @@ async def _run_commands(
...
+
+        if worktree_path is None:
+            metadata = _alembic_policy_missing_worktree_metadata(policy)
+            reason_code = "ALEMBIC_WORKTREE_REQUIRED"
+            policy_failed = True
+        else:
+            graph_result = validate_alembic_migration_chain(worktree_path, policy)
+            metadata = alembic_policy_metadata(graph_result, policy=policy)
+            reason_code = graph_result.reason_code
+            policy_failed = graph_result.status == AlembicGraphValidationStatus.failed or (
```
[the remark this answers is not in the document]
Comment (dimileeh) [reply src/awf/runtime/validation.py:480]: fixed in commit 8bb36d0b5bc457d4ce93a7d0237680a5a7999d26

File: src/awf/service/alembic_resolver.py [hunk 2]
```diff
@@ -173,90 +270,162 @@ def _safe_heads(script: ScriptDirectory) -> tuple[str, ...] | AlembicResolveResu
...
+            details.update(dict(finding.details))
+        first = findings[0]
+        return AlembicGraphValidationResult(
+            status=AlembicGraphValidationStatus.failed,
+            reason_code=first.reason_code,
             heads=_loaded_revision_heads(revisions),
-            message="Alembic revision graph is unsafe to merge automatically.",
+            message=first.message,
             details=details,
+            findings=tuple(findings),
```
[the remark this answers is not in the document]
Comment (dimileeh) [reply src/awf/service/alembic_resolver.py:387]: fixed in commit fc90903082f3dbb1d718334ef8063c3bd6d67e18

File: src/awf/profiles/models.py [hunk 3]
```diff
@@ -194,6 +194,29 @@ def _coerce_command(cls, value: object) -> object:
...
+    script_location: Annotated[str, Field(min_length=1, max_length=512)] | None = None
+    fail_on_unconfigured: bool = True
+
+    @model_validator(mode="after")
+    def _validate_paths(self) -> ProfileAlembicValidation:
+        _validate_workspace_relative_path("config_path", self.config_path)
+        if self.script_location is not None:
+            _validate_workspace_relative_path("script_location", self.script_location)
+        return self
+
```
[the remark this answers is not in the document]
Comment (dimileeh) [reply src/awf/profiles/models.py:214]: fixed in commit 266bdbcecb9e415ee2da26d4331053af35c8a904

File: src/awf/runtime/validation.py [hunk 4]
```diff
@@ -409,6 +448,79 @@ async def _run_commands(
...
+                graph_result.status == AlembicGraphValidationStatus.unsupported
+                and policy.fail_on_unconfigured
+            )
+
+        rendered = json.dumps(metadata, sort_keys=True, indent=2) + "\n"
```
[the remark this answers is not in the document]
Comment (dimileeh) [reply src/awf/runtime/validation.py]: fixed in commit 4e6e10f7e51bb3cdbe21e7f69270e9ab8061a987

File: src/awf/profiles/models.py [hunk 5]
```diff
@@ -194,6 +194,29 @@ def _coerce_command(cls, value: object) -> object:
...
+
+def _validate_workspace_relative_path(field_name: str, value: str) -> None:
+    if value.startswith("/") or ".." in value.split("/"):
```
[the remark this answers is not in the document]
Comment (dimileeh) [reply src/awf/profiles/models.py]: fixed in commit 266bdbcecb9e415ee2da26d4331053af35c8a904
